File: Java Leetcode/Solution24.java
// 24. Swap Nodes in pairs

class ListNode {
    int val;
    public ListNode next;

    ListNode() {
    }

    ListNode(int val) {
        this.val = val;
    }

    ListNode(int val, ListNode next) {
        this.val = val;
        this.next = next;
    }
}

public class Solution24 {
    public ListNode swapPairs(ListNode head) {
        ListNode dummy = new ListNode(0, head);
        ListNode prev = dummy, curr = head;

        while (curr != null && curr.next != null) {
            ListNode npn = curr.next.next;
            ListNode second = curr.next;

            second.next = curr;
            curr.next = npn;
            prev.next = second;

            prev = curr;
            curr = npn;

        }
        return dummy.next;
    }
}

File: Python Leetcode/Solution24.py
# 24. Swap Nodes in Pairs

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def swapPairs(self, head):
        dummy = ListNode(0, head)
        prev,curr = dummy, head

        while curr and curr.next:
            npn = curr.next.next
            second = curr.next

            second.next = curr
            curr.next = npn
            prev.next = second


            prev = curr
            curr = npn
        
        return dummy.next